### PR TITLE
Moar i18n: autoi18n

### PR DESCRIPTION
**Pull request by roshkins**
Check out autoi18n! It automatically i18nizes for you! Just put in the module to i18n and the language file to append to, in that order, as arguments to the program, and out pops the code ready to run (as a copy in the same directory for you to look over.) Have fun and I hope this saves time! I've already invested a day in it.

### CODE REVIEW

**Comment by andytuba** [comment]
example usage:

    node ./autoi18n.js ../lib/modules/showImages.js ../locales/locales/en.json"

[4:15 PM]

**Comment by roshkins** [comment]
It's in the utils folder, if you didn't notice.

**Comment by andytuba** [comment]
TODO:

- [ ]  tidy up merges/diff
- [ ] add `npm run` script to invoke this
- [ ] add to readme

File: utils/autoi18n.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    return '\'' + capture1.replace(/\r?\n/g, '\\n') + '\'';
+  } );
+}
+
+function parseModule(str, matchPos) {
+  var braceLocations = findBraces(str, matchPos);
+  // debugger;
+  var jsonText = str.slice(braceLocations[0], braceLocations[1] + 1);
+  var jsonTextClean = changeTicksAndEscapeNewLines(jsonText);
+  return eval("a = " + jsonTextClean); //Must be assigned to a variable for eval to return an object. JS!
```
Comment (andytuba) [inline utils/autoi18n.js]: Might also work to do `eval("(" + jsonTextClean + ")")`

File: utils/autoi18n.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      }
+    }
+  }
+  return [matchPosStart, matchPosEnd];
+}
+
+function changeTicksAndEscapeNewLines( str ) {
+  str = str.replace(/'/g, "\'");
+  return str.replace(/`([^`]*)`/g, function(match, capture1) {
+    return '\'' + capture1.replace(/\r?\n/g, '\\n') + '\'';
```
Comment (andytuba) [inline utils/autoi18n.js]: `.replace(/[\n\r]/g`

File: utils/lang.js [hunk 3]
```diff
@@ -0,0 +1 @@
...show images\" or autoExpandSelfText."},"showImagesOptionsAutoExpandSelfTextDescription":{"message":"When loading selftext from an Aa+ expando, auto expand enclosed expandos."},"showImagesOptionsAutoExpandSelfTextFirstVisibleNonMutedDescription":{"message":"In selftexts, expand the first visible potentially non-muted expando."},"showImagesOptionsAutoExpandSelfTextNSFWDescription":{"message":"Also expand expandos in selftexts which are marked NSFW."},"showImagesOptionsShowSiteAttributionDescription":{"message":"Show the site logo and name after embedded content."},"showImagesOptionsExpandoCommentRedirectsDescription":{"message":"How should RES handle posts where the link is redirected to the comments page with preview expanded?"},"showImagesOptionsShowVideoControlsDescription":{"message":"Show controls such as pause/play, step and playback rate."},"showImagesOptionsOnlyPlayMutedWhenVisibleDescription":{"message":"Auto-pause muted videos when they are not visible."},"showImagesOptionsMaxSimultaneousPlayingDescription":{"message":"Auto-play at most this many muted videos simultaneously. (0 for no limit)"},"showImagesOptionsAutoplayVideoDescription":{"message":"Autoplay inline videos"}}
```
Comment (andytuba) [inline utils/lang.js]: Looks like all this should be appended to the json inside `en.json`?

**Pull request closed without merging**